lib/ws_servers/api/index.js

```diff
@@ -92,20 +98,48 @@ module.exports = class APIWSServer extends WSServer {
     send(ws, ['info.version', VERSION])
     send(ws, ['info.exchanges', EXCHANGE_ADAPTERS.map(({ id }) => id)])
 
-    await PI.forEach(EXCHANGE_ADAPTERS, async ({ id }) => {
-      const { Market } = this.db
-      const markets = await Market.find([['exchange', '=', id]])
-
-      send(ws, ['info.markets', id, markets])
-    }).catch((err) => {
-      this.d('error sending markets to client: %s', err.stack)
-    })
+    for (const { id } of EXCHANGE_ADAPTERS) {
+      try {
+        const { Market } = this.db
+        const markets = await Market.find([['exchange', '=', id]])
 
+        send(ws, ['info.markets', id, markets])
+      } catch (err) {
+        this.d('error sending markets to client: %s', err.stack)
+      }
+    }
     const credentials = await getCredentials(this.db)
 
     send(ws, ['info.auth_configured', !!credentials])
   }
 
+  async sendActiveAlgoOrders (ws) {
+    const { UserSettings } = this.db
+    const { AlgoOrder } = this.algoDB
+    const { userSettings: settings } = await UserSettings.getAll()
+    const { dms } = settings || DEFAULT_SETTINGS
+
+    if (dms) {
+      this.as.open()
```

Comment:
afaik we should also ask the user, if the dms option is not set, just to make sure